eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Tick.hs

```diff
@@ -163,6 +168,64 @@ validatingTickTransition nes slot = do
 
   pure $ nes' {nesEs = es''}
 
+-- | This is a limited version of 'validatingTickTransition' which is only suitable
+-- for the future ledger view.
+validatingTickTransitionFORECAST ::
+  forall tick era.
+  ( State (tick era) ~ NewEpochState era,
+    State (Core.EraRule "PPUP" era) ~ PPUPState era,
+    UsesPParams era,
+
+    BaseM (tick era) ~ ShelleyBase,
+    STS (tick era)
+  ) =>
+  NewEpochState era ->
+  SlotNo ->
+  TransitionRule (tick era)
+validatingTickTransitionFORECAST nes slot = do
+  -- This whole function is a specialization of an inlined 'NEWEPOCH'.
+  --
+  -- The ledger view is built entirely from the 'nesPd' and 'esPP' and
+  -- '_genDelegs', so those are the only pieces we bother to update here.
+
+  epoch <- liftSTS $ do
+    ei <- asks epochInfoPure
+    epochInfoEpoch ei slot
+
+  let es = nesEs nes
+      EpochState
+        { esSnapshots = ss,
+          esLState = ls,
+          esPp = pp
+        } = es
+
+  -- the relevant 'NEWEPOCH' logic
+  let pd' = _pstakeMarkPoolDistr ss
+  if unEpochNo epoch /= unEpochNo (nesEL nes) + 1 then pure nes else do
+
+    -- We can skip 'SNAP'; we already have the equivalent pd'.
+
+    -- We can skip 'MIR', 'POOLREAP', 'UPEC' (which calls 'NEWPP'); we don't
+    -- need to do the checks: if the checks would fail, then the node will fail
+    -- in the 'TICK' rule if it ever tries to validate blocks for which the
+    -- return value here was used to validate their headers.
+
+    coreNodeQuorum <- liftSTS $ asks quorum
+
+    let pp' =
+            maybe pp id
+          $ votedValue
```

Comment:
just for posterity, let me mention that just because the core nodes reach quorum on protocol parameter update doesn't mean it will be accepted. At the epoch boundary, `UPEC` does some final sanity checks. (This turns out to be problematic, see [ADR-3](https://github.com/input-output-hk/cardano-ledger/pull/3190/files#diff-fb71f1e0f9d8938e0f10af3f59332fa73b3c539fbb4fc4065c623b23327a12b3R56) and will be changing in the next ledger era.

hooray for property tests! :tada: